addons/account/models/account_move.py

```diff
@@ -1196,7 +1196,7 @@ def _compute_tax_totals(self):
 
                         kwargs['base_lines'].append(self.env['account.tax']._convert_to_tax_base_line_dict(
                             None,
-                            partner=move.partner_id,
+                            partner=move.partner_id.commercial_partner_id,
```

Comment:
This is not the root cause of your issue. The problem here is far more complicated because it's an inconsistency between the account_move taxes grouping key and the one in account_tax.py